crates/synthesis/src/linalg/cos_sin_decomp.rs

```diff
@@ -228,17 +175,37 @@ pub fn cos_sin_decomposition(u: DMatrix<Complex64>) -> CosSinDecompReturn {
     (l0, l1, r0, r1, thetas)
 }
 
-// TODO: Remove this function and all the python interface when QSD is ported to rust
+fn faer_mat_to_ndarray(faer_mat: &Mat<Complex64>) -> Array2<Complex64> {
+    let rows = faer_mat.nrows();
+    let cols = faer_mat.ncols();
+
+    let mut ndarray_mat = Array2::<Complex64>::default((rows, cols));
+
+    for i in 0..rows {
+        for j in 0..cols {
+            ndarray_mat[[i, j]] = faer_mat[(i, j)];
+        }
+    }
+
+    ndarray_mat
+}
+
 #[pyfunction]
 pub fn cossin<'py>(py: Python<'py>, u: PyReadonlyArray2<Complex64>) -> PyResult<Bound<'py, PyAny>> {
     let array = u.as_array();
     let shape = array.shape();
-    let mat: DMatrix<Complex64> = DMatrix::from_fn(shape[0], shape[1], |i, j| array[[i, j]]);
-    let res = cos_sin_decomposition(mat);
+    let mat: Mat<Complex64> = Mat::from_fn(shape[0], shape[1], |i, j| array[[i, j]]);
+    let res = cos_sin_decomposition(mat.as_ref());
+
+    let arr0 = faer_mat_to_ndarray(&res.0);
+    let arr1 = faer_mat_to_ndarray(&res.1);
+    let arr2 = faer_mat_to_ndarray(&res.2);
+    let arr3 = faer_mat_to_ndarray(&res.3);
+
     Ok((
-        (res.0.to_pyarray(py), res.1.to_pyarray(py)),
+        (arr0.to_pyarray(py), arr1.to_pyarray(py)),
```

Comment:
Is there a better way than the above two-step process: first converting a `faer` matrix to a `ndarray` one, and the converting that to `py_array`?